File: src/main/java/indi/midreamsheep/app/tre/model/shortcut/TREShortcutKey.java
package indi.midreamsheep.app.tre.model.shortcut;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.Set;

@Data
@AllArgsConstructor
public class TREShortcutKey {
    private Long[] keys;

    public TREShortcutKey(long... keyCode) {
        this.keys = new Long[keyCode.length];
        for (int i = 0; i < keyCode.length; i++) {
            this.keys[i] = keyCode[i];
        }
    }

    public boolean check(Set<Long> pressedKeys){
        return pressedKeys.containsAll(Set.of(keys))&&keys.length==pressedKeys.size();
    }
}

File: src/main/kotlin/indi/midreamsheep/app/markdown/model/shortcut/TREShortcutKey.kt
package indi.midreamsheep.app.markdown.model.shortcut

import androidx.compose.ui.input.key.KeyEvent
import indi.midreamsheep.app.markdown.context.TREContext

interface TREShortcutKey{
    fun action(context:TREContext)
    fun getKeys():List<List<Long>>
}